infoportal-client/src/features/Meta/useMetaData.ts

```diff
@@ -100,25 +100,33 @@ export const useMetaDashboardData = (data: Seq<IKoboMeta>) => {
   const [customFilters, setCustomFilters] = usePersistentState<MetaDashboardCustomFilter>({}, {storageKey: 'meta-dashboard-custom-filters'})
   const distinctBy = useMemo(() => new Set(customFilters.distinctBy), [customFilters.distinctBy])
 
+  const updatedDataWithBlankStatus = data.map(d => ({
+    ...d,
+    status: d.status ?? KoboMetaStatus.Blank
+  }))
+
   const filteredData = useMemo(() => {
-    const filteredBy_date = data.filter(d => {
-      try {
-        const isDateIn = PeriodHelper.isDateIn(period, d.date)
-        if (!isDateIn) return false
-        const isDateCommitIn = (!periodCommit.start && !periodCommit.end)
-          || PeriodHelper.isDateIn(periodCommit, d.lastStatusUpdate ?? d.date) && d.status === KoboMetaStatus.Committed
-        if (!isDateCommitIn) return false
+    const filteredByDateAndCommit = updatedDataWithBlankStatus.filter(d => {
+      const isDateIn = PeriodHelper.isDateIn(period, d.date)
+      const isDateCommitIn = (!periodCommit.start && !periodCommit.end) ||
+        (PeriodHelper.isDateIn(periodCommit, d.lastStatusUpdate ?? d.date) && d.status === KoboMetaStatus.Committed)
+      return isDateIn && isDateCommitIn
+    })
+
+    const filteredByStatus = filteredByDateAndCommit.filter(d => {
```

Comment:
Hmm it's working but you are re-inventing the wheel. The mechanism should already be implemented in `DataFilter.filterData`.